samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -188,98 +183,82 @@ private void validateConfig() {
-    // mapping from a source stream to all join specs reachable from it
-    Multimap<JoinOperatorSpec, StreamEdge> joinSpecToStreamEdges = HashMultimap.create();
-    // reverse mapping of the above
-    Multimap<StreamEdge, JoinOperatorSpec> streamEdgeToJoinSpecs = HashMultimap.create();
-    // A queue of joins with known input partitions
-    Queue<JoinOperatorSpec> joinQ = new LinkedList<>();
-    // The visited set keeps track of the join specs that have been already inserted in the queue before
-    Set<JoinOperatorSpec> visited = new HashSet<>();
-
-    jobGraph.getSpecGraph().getInputOperators().forEach((streamId, inputOperatorSpec) -> {
-        StreamEdge streamEdge = jobGraph.getOrCreateStreamEdge(getStreamSpec(streamId, streamConfig));
-        // Traverses the StreamGraph to find and update mappings for all Joins reachable from this input StreamEdge
-        findReachableJoins(inputOperatorSpec, streamEdge, joinSpecToStreamEdges, streamEdgeToJoinSpecs, joinQ, visited);
-      });
-
-    // At this point, joinQ contains joinSpecs where at least one of the input stream edge partitions is known.
-    while (!joinQ.isEmpty()) {
-      JoinOperatorSpec join = joinQ.poll();
-      int partitions = StreamEdge.PARTITIONS_UNKNOWN;
-      // loop through the input streams to the join and find the partition count
-      for (StreamEdge edge : joinSpecToStreamEdges.get(join)) {
-        int edgePartitions = edge.getPartitionCount();
-        if (edgePartitions != StreamEdge.PARTITIONS_UNKNOWN) {
-          if (partitions == StreamEdge.PARTITIONS_UNKNOWN) {
-            //if the partition is not assigned
-            partitions = edgePartitions;
-            log.info("Inferred the partition count {} for the join operator {} from {}."
-                , new Object[] {partitions, join.getOpId(), edge.getName()});
-          } else if (partitions != edgePartitions) {
-            throw  new SamzaException(String.format(
-                "Unable to resolve input partitions of stream %s for the join %s. Expected: %d, Actual: %d",
-                edge.getName(), join.getOpId(), partitions, edgePartitions));
-          }
-        }
-      }
-
-      // assign the partition count for intermediate streams
-      for (StreamEdge edge : joinSpecToStreamEdges.get(join)) {
-        if (edge.getPartitionCount() <= 0) {
-          log.info("Set the partition count to {} for input stream {} to the join {}.",
-              new Object[] {partitions, edge.getName(), join.getOpId()});
-          edge.setPartitionCount(partitions);
-
-          // find other joins can be inferred by setting this edge
-          for (JoinOperatorSpec op : streamEdgeToJoinSpecs.get(edge)) {
-            if (!visited.contains(op)) {
-              joinQ.add(op);
-              visited.add(op);
-            }
-          }
-        }
-      }
+  private void validateJoinInputStreamPartitions(JobGraph jobGraph) {
```

Comment:
I feel the essence of this method is already captured in the method name and having to suffix `joined` in declaration/method names/comments within seems redundant and in some cases confusing. 

E.g. `joinedStreamGroups` - it is not obvious or intuitive as to what this means.

What do you think?

Reply:
Please, mark every such case individually. I can relate to what you're saying to some extent in the case of `JoinedStreamGroup` but I don't see many other cases.

As for `joinedStreamGroups`, would you prefer `joinedStreamEdges`? I don't have a strong preference but I don't think `streamEdges` would be sufficiently clear. Please, let me know what you think.